crates/flox/src/commands/environment.rs

```diff
@@ -422,50 +388,45 @@ impl Install {
     pub async fn handle(self, flox: Flox) -> Result<()> {
         subcommand_metric!("install");
 
-        let packages: Vec<_> = self
-            .packages
-            .iter()
-            .map(|package| FloxPackage::parse(package, &flox.channels, DEFAULT_CHANNEL))
-            .collect::<Result<Vec<_>, _>>()?
-            .into_iter()
-            .dedup()
-            .collect();
-
+        debug!(
+            "installing packages [{}] to {:?}",
+            self.packages.as_slice().join(", "),
+            self.environment
+        );
         let mut environment = self
             .environment
             .to_concrete_environment(&flox)?
             .into_dyn_environment();
-
-        // todo use set?
-        // let installed = environment
-        //     .packages(&flox.nix(Default::default()), &flox.system)
-        //     .await?
-        //     .into_iter()
-        //     .map(From::from)
-        //     .collect::<Vec<FloxPackage>>();
-
-        // if let Some(installed) = packages.iter().find(|pkg| installed.contains(pkg)) {
-        //     anyhow::bail!("{installed} is already installed");
-        // }
-
-        let packages_str = packages_to_string(&packages);
-        let plural = packages.len() > 1;
-
-        if environment
-            .install(packages, &flox.nix(Default::default()), flox.system)
-            .await
-            .context("could not install packages")?
-        {
-            println!(
-                "✅ Installed {packages_str} into '{}' environment.",
-                environment.environment_ref()
-            );
+        let nix = flox.nix::<NixCommandLine>(vec![]);
+        let installation = environment
+            .install(self.packages.clone(), &nix, flox.system.clone())
+            .await?;
+        if installation.new_manifest.is_some() {
+            // Write the new manifest
+            match self.environment {
+                EnvironmentSelect::Dir(dir) => {
+                    let manifest_path = dir.join(".flox/env").join(MANIFEST_FILENAME);
+                    debug!("writing new manifest to {}", manifest_path.display());
+                    std::fs::write(
+                        dir.join(".flox/env").join(MANIFEST_FILENAME),
+                        &installation.new_manifest.unwrap(),
+                    )?;
+                },
+                EnvironmentSelect::Remote(_) => {
+                    warn!("remote environments not supported yet");
+                    // TODO: handle remote environments
+                },
+            }
```

Comment:
suggestion: move this back to the environment implementation.
file layout etc should be encapsulated within the trait impl

Reply:
I addressed this comment in the `path_environment` impl